packages/elements-core/src/components/Docs/Docs.tsx

```diff
@@ -46,18 +25,40 @@ export interface IDocsComponentProps<T = unknown> {
   uri?: string;
 }
 
-export const Docs = React.memo<IDocsProps>(({ nodeType, nodeData, className, headless, uri }) => {
-  const parsedNode = useParsedData(nodeType, nodeData);
+export interface DocsProps extends BaseDocsProps {
+  unparsedNode?: UnparsedNode;
+  node?: ParsedNode;
+  useNodeForRefResolving?: boolean;
+}
+
+export interface DocsComponentProps<T = unknown> extends BaseDocsProps {
+  /**
+   * The input data for the component to display.
+   */
+  data: T;
+}
+
+export const Docs = React.memo<DocsProps>(({ unparsedNode, node, useNodeForRefResolving = false, ...commonProps }) => {
+  const parsedNode = useParsedData(unparsedNode, node);
 
   if (!parsedNode) {
     // TODO: maybe report failure
     return null;
   }
+  const parsedDocs = <ParsedDocs node={parsedNode} {...commonProps} />;
+
+  if (useNodeForRefResolving) {
+    return <InlineRefResolverProvider document={parsedNode.data}>{parsedDocs}</InlineRefResolverProvider>;
+  }
```

Comment:
Any chance we can just take the inline ref resolver context and making the condition `if(!context)`? Assuming if someone has a context around they want that, otherwise we fall back to this? May be a stupid idea, just throwing it out here.

Reply:
I was thinking about this, but it seemed to high-concept to pass a review :joy: 

Maybe do it in a separate PR? We still have an issue to fix problems with providers.